src/provider/cloud-element-templates/cmd/ChangeElementTemplateHandler.js

```diff
@@ -279,7 +281,21 @@ export default class ChangeElementTemplateHandler {
       // (2) update old inputs and outputs
       if (oldProperty && oldInputOrOutput) {
 
+        // (2a) exclude old inputs and outputs from cleanup (unless optional)
+        if (
+          shouldUpdate(newPropertyValue, newProperty) ||
+          (isOptional && propertyChanged(oldInputOrOutput, oldProperty))
```

Comment:
Trying to wrap my head around what's checked here. My assumption would be we always want to update when the property changes and when it is not optional.

I'm wondering if `isOptional` is needed here. `shouldUpdate` returns `false` when the value is empty and optional. For the second condition, `isOptional` should always be true when we reach it, or am I missing something?